asset/asset.go

```diff
@@ -22,12 +23,20 @@ import (
 // compressed, 33-byte form.
 type SerializedKey [33]byte
 
+// SchnorrKey is a type for representing an x-only public key, serialized in the
+// 32-byte form.
+type SchnorrKey [32]byte
+
 // SchnorrSerialized returns the Schnorr serialized, x-only 32-byte
 // representation of the serialized key.
 func (s SerializedKey) SchnorrSerialized() []byte {
 	return s[1:]
 }
 
+func (s SchnorrKey) SchnorrSerialized() []byte {
```

Comment:
Should we start to make this into a slim interface? Or are the type defs above sufficient?

Reply:
Not sure what you mean wrt. slim interface - I think using the `any` type would cause trouble elsewhere. Maybe we can do something with a custom type constraint though?

Reply:
Mean moreso making `SchnorrSerialized` into an interface and using that elsewhere. Then either key can be passed in depending on context. I think we want to have this be explicit though, so we can ensure we're using the correct serialization when necessary.